tests/v2/validation/networking/connectivity/network_policy_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/sshkeys"
+	shepworkloads "github.com/rancher/shepherd/extensions/workloads"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/url"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type NetworkPolicyTestSuite struct {
+	suite.Suite
+	session     *session.Session
+	client      *rancher.Client
+	project     *management.Project
+	clusterName string
+	namespace   *v1.SteveAPIObject
+	steveClient *v1.Client
+}
+
+const (
+	nodeRole = "control-plane"
+	// Ping once
+	pingCmd        = "ping -c 1"
+	successfulPing = "0% packet loss"
+)
+
+func (n *NetworkPolicyTestSuite) TearDownSuite() {
+	n.session.Cleanup()
+}
+
+func (n *NetworkPolicyTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	n.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(n.T(), err)
+
+	n.client = client
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmpty(n.T(), clusterName, "Cluster name to install is not set")
+	n.clusterName = clusterName
+
+	cluster, err := clusters.NewClusterMeta(client, clusterName)
+	require.NoError(n.T(), err)
+
+	projectConfig := &management.Project{
+		ClusterID: cluster.ID,
+		Name:      pingPodProjectName,
+	}
+
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, pingPodProjectName)
+	n.project = createdProject
+
+	names := newNames()
+
+	namespaceName := names.random["namespaceName"]
+	daemonsetName := names.random["daemonsetName"]
+
+	n.T().Logf("Creating namespace with name [%v]", namespaceName)
+	n.namespace, err = namespaces.CreateNamespace(n.client, namespaceName, "{}", map[string]string{}, map[string]string{}, n.project)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), n.namespace.Name, namespaceName)
+
+	n.steveClient, err = n.client.Steve.ProxyDownstream(n.project.ClusterID)
+	require.NoError(n.T(), err)
+
+	testContainerPodTemplate := newPodTemplateWithTestContainer()
+
+	n.T().Logf("Creating a daemonset with the test container with name [%v]", daemonsetName)
+	daemonsetTemplate := shepworkloads.NewDaemonSetTemplate(daemonsetName, n.namespace.Name, testContainerPodTemplate, true, nil)
+	createdDaemonSet, err := n.steveClient.SteveType(workloads.DaemonsetSteveType).Create(daemonsetTemplate)
+	require.NoError(n.T(), err)
+	assert.Equal(n.T(), createdDaemonSet.Name, daemonsetName)
+
+	n.T().Logf("Waiting for daemonset [%v] to have expected number of available replicas", daemonsetName)
+	err = charts.WatchAndWaitDaemonSets(n.client, n.project.ClusterID, n.namespace.Name, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+}
+
+func (n *NetworkPolicyTestSuite) TestPingPods() {
```

Comment:
can you add something to make the test name more clear, i.e. `TestPingPodsFromCPNode`

Reply:
sure